students/scott_bromley/lesson04/trigrams.py

```diff
@@ -0,0 +1,66 @@
+#!/usr/bin/env python3
+import sys
+import random
+
+
+def read_in_data(filename):
+    """
+    read in filename
+    :param filename: name of .txt file
+    :return: string of filename input
+    """
+    with open(filename, 'r') as f:
+        translation_dict = {9: ' ', 10: ' ', 13: ' ', 33: None, 40: None, 41: None, 44: None, 45: None, 46: None,
+                            59: None}
+        text = f.read().translate(translation_dict)
+        return text
+
+
+def make_words(text):
+    """
+    creates a list of all words in text
+    :param text: string of input
+    :return: list of words in text
+    """
+    words = text.split()
+    return words
+
+
+def build_trigram(words):
+    """
+    build trigram out of words
+    :param words: list of words
+    :return: dict using tuple of word pairs as key and dict as values
+    """
+    trigram = {}
+    for i in range(len(words) - 2):
+        trigram.setdefault(tuple(words[i:i + 2]), words[i + 2])
```

Comment:
>  I cannot get the values to update with additional matches

you're never appending values to your dictionary here; the signature should be: 
```
newdata.setdefault(<key>, []).append(<value>)
```
I prefer to use `defaultdict`:
```
from collections import defaultdict
trigrams = defaultdict(list)

....
trigrams[key].append(value)
```